packages/lib/server/getServerErrorFromUnknown.ts

```diff
@@ -34,37 +38,71 @@ function parseZodErrorIssues(issues: ZodIssue[]): string {
 }
 
 export function getServerErrorFromUnknown(cause: unknown): HttpError {
+  let traceId: string | undefined;
+  let tracedData: Record<string, unknown> | undefined;
+
+  if (cause instanceof TracedError) {
+    traceId = cause.traceId;
+    tracedData = cause.data;
+    cause = cause.originalError;
+  }
+
+  if (cause instanceof TRPCError) {
+    const statusCode = getHTTPStatusCodeFromError(cause);
+    return new HttpError({
+      statusCode,
+      message: cause.message,
+      data: traceId ? { ...tracedData, traceId } : undefined,
+    });
+  }
   if (isZodError(cause)) {
     return new HttpError({
       statusCode: 400,
       message: parseZodErrorIssues(cause.issues),
       cause,
+      data: traceId ? { ...tracedData, traceId } : undefined,
     });
   }
   if (cause instanceof SyntaxError) {
     return new HttpError({
       statusCode: 500,
       message: "Unexpected error, please reach out for our customer support.",
+      data: traceId ? { ...tracedData, traceId } : undefined,
     });
   }
   if (isPrismaError(cause)) {
-    return getServerErrorFromPrismaError(cause);
+    const prismaError = getServerErrorFromPrismaError(cause);
```

Comment:
Similary getServerErrorFromPrismaError is also creating HttpError and then soon after we create another HttpError